extensions/amp-analytics/0.1/vendors/snowplow_v2.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {jsonLiteral} from '../../../../src/json';
+
+const SNOWPLOW_V2_CONFIG = jsonLiteral({
+  'linkers': {
+    'linker': {
+      'ids': {
+        '_sp_duid': '${duid}',
+      },
+    },
+  },
+  'cookies': {
+    'enabled': true,
+    'cookieMaxAge': 1800,
+    '_sp_duid': {
+      'value':
+        '$IF($IF(COOKIE(_sp_duid), COOKIE(_sp_duid), LINKER_PARAM(linker,_sp_duid)), $IF(COOKIE(_sp_duid), COOKIE(_sp_duid), LINKER_PARAM(linker,_sp_duid)), $SUBSTR(QUERY_PARAM(_sp),0,36))',
+    },
+  },
+  'vars': {
+    'aaVersion': 'amp-1.0.0',
+    'ampVistorId': 'CLIENT_ID(_sp_id)',
+    'nullString': 'null',
+    'customEventTemplate': [
+      '{',
+      '"schema":"iglu:com.snowplowanalytics.snowplow/unstruct_event/jsonschema/1-0-0",',
+      '"data":{',
+      '"schema":"iglu:${customEventSchemaVendor}/${customEventSchemaName}/jsonschema/${customEventSchemaVersion}",',
+      '"data":{',
+      '${customEventSchemaData}',
```

Comment:
This feels really weird, having to write escaped 'JSON' without the wrapping curly braces. Is there a specific reason for this? I can only imagine this being harder to generate than including the braces to begin with.

Reply:
Yeah I agree, it's janky.

There is a reason - similar to the other comment, I couldn't get anything else to work. For example, attempts to set the var as a JSON object and using JSON.stringify() outputs `"[object Object]"`.

Reply:
(If that is indeed what you meant, feel free to either resolve this convo or comment with further feedback. :) )

Reply:
Yep, just meant the braces bit, apologies for my obtuse writing there!

Reply:
No was entirely in my reading of it, was just hard to get my head out of the other problem space. Resolving this now as I've made that change!